puppet/spec/unit/reports/puppetdb_spec.rb

```diff
@@ -103,6 +104,18 @@ def without_producer_timestamp(json_body)
       result["producer"].should == "foo"
     end
 
+    it "should include noop_pending or nil" do
+      if defined?(subject.noop_pending) then
+        subject["noop_pending"] = false
```

Comment:
`subject.noop_pending` is read only (attr_reader). Great, if you can workaround it this way.